### ISSUE TITLE
How to run on Android

### ISSUE THREAD

**Issue by elie222** [spencercarli/react-native-meteor-boilerplate#41]
This used to work for me in the past, but I'm now I'm struggling to get it going.

The error I get after running `react-native run-android` is:

```
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```

I don't have an emulator running, but I'm not sure how to either. In the past, just running this command would load up the emulator.

**Comment by spencercarli** [comment]
Has anything changed about your development environment? Make sure that your machine is [properly configured](http://facebook.github.io/react-native/releases/0.23/docs/android-setup.html) for android development.

**Comment by abelovic** [comment]
I am also having an issue (although not the same as Elie222).
- I can run the React-Native AwesomeProject correctly however when I run this project it says the following:

"RNApp has stopped. open app again"

When I click to refresh (reopen it) nothing happens. I don't see any errors in my build ouput but here it is anyways:

C:\meteor\react-native-meteor-boilerplate\RNApp>react-native run-android
JS server already running.
Running C:\Users\myusername\AppData\Local\Android\sdk/platform-tools/adb reverse tcp:8081 tcp:8081
Building and installing the app on the device (cd android && gradlew.bat installDebug...
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
:app:checkDebugManifest
:app:preReleaseBuild UP-TO-DATE
:app:prepareComAndroidSupportAppcompatV72301Library UP-TO-DATE
:app:prepareComAndroidSupportRecyclerviewV72301Library UP-TO-DATE
:app:prepareComAndroidSupportSupportV42321Library UP-TO-DATE
:app:prepareComFacebookFrescoDrawee0110Library UP-TO-DATE
:app:prepareComFacebookFrescoFbcore0110Library UP-TO-DATE
:app:prepareComFacebookFrescoFresco0110Library UP-TO-DATE
:app:prepareComFacebookFrescoImagepipeline0110Library UP-TO-DATE
:app:prepareComFacebookFrescoImagepipelineBase0110Library UP-TO-DATE
:app:prepareComFacebookFrescoImagepipelineOkhttp30110Library UP-TO-DATE
:app:prepareComFacebookReactReactNative0300Library UP-TO-DATE
:app:prepareOrgWebkitAndroidJscR174650Library UP-TO-DATE
:app:prepareDebugDependencies
:app:compileDebugAidl UP-TO-DATE
:app:compileDebugRenderscript UP-TO-DATE
:app:generateDebugBuildConfig UP-TO-DATE
:app:generateDebugAssets UP-TO-DATE
:app:mergeDebugAssets UP-TO-DATE
:app:generateDebugResValues UP-TO-DATE
:app:generateDebugResources UP-TO-DATE
:app:mergeDebugResources UP-TO-DATE
:app:bundleDebugJsAndAssets SKIPPED
:app:processDebugManifest UP-TO-DATE
:app:processDebugResources UP-TO-DATE
:app:generateDebugSources UP-TO-DATE
:app:processDebugJavaRes UP-TO-DATE
:app:compileDebugJavaWithJavac UP-TO-DATE
:app:compileDebugNdk UP-TO-DATE
:app:compileDebugSources UP-TO-DATE
:app:preDexDebug UP-TO-DATE
:app:dexDebug UP-TO-DATE
:app:validateDebugSigning
:app:packageDebug UP-TO-DATE
:app:zipalignDebug UP-TO-DATE
:app:assembleDebug UP-TO-DATE
:app:installDebug
Installing APK 'app-debug.apk' on 'emulator-5554 - 7.0'
Installed on 1 device.

BUILD SUCCESSFUL

Total time: 9.222 secs
Starting the app on emulator-5554 (C:\Users\myusername\AppData\Local\Android\sdk/platform-tools/adb -s emulator-5554 shell am start -n com.rnapp/.MainActivity)...
Starting: Intent { cmp=com.rnapp/.MainActivity }

C:\meteor\react-native-meteor-boilerplate\RNApp>

My AVD is a Galaxy S7 Edge skin running Android 7, API 24 on Intel Atom (x86_64). Tried other AVDs with lower Android versions and they did not work either.

According to the instructions "In RNApp/app/config.js change localhost to your machine's IP address"

I did not find a config.js file but did change "RNApp/app/config/settings.js" and changed localhost to my machine's IP.

I am running this with the latest Android Studio installed on Windows 10. I have JDK8 installed and my environment system variables have ANDROID_TOOLS, ANDROID_PLATFORM_TOOLS pointed to the the sdk platform-tools location and ANDROID_HOME is pointed to the sdk root

Any idea what is wrong?

**Comment by acomito** [comment]
Do you have the most up to date mac OS?

**Comment by abelovic** [comment]
I tried the above on Windows 10. 

Interestingly I just tried this on the IOS Simulator on a Mac (El Capitan 10.11.6) and it works correctly (other than the react 15.3 warnings which should be fixed but is unrelated).

The Android Simulator on Windows does not seem to work or I am doing something wrong :(

**Comment by spencercarli** [comment]
@abelovic have you tried the android simulator on a mac? if so did that work?

@elie222 are you using mac or windows? There's another windows issue in #36... wonder if it's something w/ RN in that environment?

**Comment by spencercarli** [comment]
Also I updated the docs to point to config/settings.js - thanks for pointing that out @abelovic

**Comment by elie222** [comment]
So I tried a bunch of stuff and not working for me yet. Just been
developing on iOS simulator for now.

Will google it more tomorrow to try figure it out.

**Comment by abelovic** [comment]
@spencercarli Ok I finally got Android Studio working on a Mac. Exactly the same problem :( The AwesomeProject template from Facebook works perfectly fine on both Windows and Mac but this boilerplate app fails on Android with no errors reported during the build process. I am assuming there must be something wrong with the boilerplate Android build files?

Has anyone gotten this to work on an Android emulator on any OS?

@elie222 You don't need to start an emulator in IOS like you do in Android because it is already built in and runs when the xcode starts. To get this to work on Android you need to setup an AVD and start the emulator. Although you can create one from the SDK Manager (from your tools folder in your installation directory), I personally use Android Studio because the interface allows for more customization options i.e. setting a skin etc... Once you have the SDK properly configured you should see the following inside Android Studio:

Tools > Android > AVD Manager

A wizard will popup and you can follow the instructions on creating a new virtual device. Just google the device name that you want to target and you should find the skin and settings for that device (or you can just use one of their presets). Make sure you set the correct CPU and if you have Intel for instance use Intel HXAM driver or it will be really slow. Unfortunately this driver doesn't work in a VM if you are using one.

Once this is complete just run the emulator (from the list of devices) and then run the 'react-native run-android' from the command line.

Then you should be where I am at will most likely see the same issues I am running into ;)

**Comment by abelovic** [comment]
Also just tried this using GenyMotion on Windows - AVD Galaxy S6 6.0.0 API 23

and same thing - AwesomeProject works but can't get this project to work on Android :(

**Comment by abelovic** [comment]
Update: 

I got this to work by re-running 

`react-native init RNApp`

and copying the code from index.android.js back in manually. Maybe a file didn't get committed or maybe the name was changed for this project manually and not updated everywhere? Not sure but either way it appears there was something wrong with the Android files (IOS worked great).

Also if you are running this on Windows 10 like I am, you may run into issues with the accounts-password package when trying to upgrade this to meteor 1.4. I was able to get this to work by installing the windows build tools:

`npm i --global --production windows-build-tools`

and then installing the Visual Studio C++ SDK (not installed by default) which is apparently needed to get the bcrypt dependency to work. For more details see: [#7469](https://github.com/meteor/meteor/issues/7469)

Thanks for this boilerplate - this is a great starting point on how to get up and running with react-native and meteor!

**Comment by tororudy** [comment]
Hi @spencercarli , I have the same issue as @abelovic when running on android device. I haven't test it yet on IOS device, only test it on IOS simulator and it works great! 

Hi @abelovic, can you explain more on how you got this to work by re-running `react-native init RNApp` and the code from index.android.js back in manually? Thanks.

**Comment by abelovic** [comment]
Here are the steps I used: 
1. Delete the RNApp folder. _Note:_ I'm guessing if we can figure out which files are incorrect we probably can just fix those instead of deleting the entire RNApp folder. This was just easier/faster for me.
2. Run `react-native init RNApp` from the root of the project which will create an RNApp folder at the same level as MeteorApp folder
3. Copy the app folder, index.android.js, index.ios.js, and package.json from the boilerplate back into your new RNApp folder
4. Run `npm install`
5. Make sure you configure the RNApp/app/config/settings.js as per the boilerplate documentation
6. Run `meteor` in a different command/bash shell from the MeteorApp folder
7. Start your android emulator
8. Run `react-native run-android`

I am running on Windows 10 and this works for me on all emulators i.e. GenyMotion, Android Studio etc... with no errors :)

**Comment by tororudy** [comment]
@abelovic thanks for your elaboration on steps that you used. I understand that your steps may not fix the issue, I just curious how this boilerplate running in my android device since in ios simulator it works great 😄 .

I was trying to run on real android device (not an emulator) and get more or less the same message as you reported which is "Unfortunately, RNApp has stopped". I follow the steps that you post but with no success and still get the same error message on android device. I'm running this on Mac (El Capitan 10.11.6). Have you try to run this on real android device?

**Comment by abelovic** [comment]
@tororudy Unfortunately I have an iPhone not an android device so I can't try it out :(

Did you try an emulator? I wouldn't think they would behave different :/ 

I will also double check my steps to make sure I didn't miss anything

**Comment by tororudy** [comment]
Hi @abelovic , I haven't try android emulator yet. I'll try it and let you know the outcome soon.

**Comment by ghost** [comment]
Hi, I have the exactly problem running on Android Simulator, it immediately  stops working after launched.

**Comment by spencercarli** [comment]
Anyone want to give the [`android-fix` branch](https://github.com/spencercarli/react-native-meteor-boilerplate/tree/android-fix) a shot?

**Comment by ghost** [comment]
@spencercarli  android-fix branch works like charm, Thanks!
I've tried running on simulators, not on physical device yet.

**Comment by spencercarli** [comment]
Closed via #52
